Tests/L0/Maven/_suite.ts

```diff
@@ -4,11 +4,99 @@
 import path = require('path');
+import fs = require('fs');
+
+import tr = require('../../lib/vsts-task-lib/toolRunner');
+
+import pmd = require('../../../Tasks/Maven/mavenpmd');
+import ca = require('../../../Tasks/Maven/mavencodeanalysis');
+import ar = require('../../../Tasks/Maven//analysisResult');
 
 function setResponseFile(name: string) {
     process.env['MOCK_RESPONSES'] = path.join(__dirname, name);
 }
 
+// Sets up a Maven TaskRunner instance with all of the required default settings
+function setupDefaultMavenTaskRunner():trm.TaskRunner {
+    var taskRunner = new trm.TaskRunner('Maven', true, true);
+    // default required settings
+    taskRunner.setInput('mavenVersionSelection', 'Path');
+    taskRunner.setInput('mavenPath', '/home/bin/maven') // Make that checkPath returns true for this filename in the response file
+    taskRunner.setInput('goals', 'package');
+    taskRunner.setInput('javaHomeSelection', 'JDKVersion');
+    taskRunner.setInput('jdkVersion', 'default');
+    taskRunner.setInput('jdkArchitecture', 'x86');
+    taskRunner.setInput('testResultsFiles', '**/TEST-*.xml');
+    taskRunner.setInput('sqAnalysisEnabled', 'false');
+    taskRunner.setInput('mavenPOMFile', 'pom.xml');
+
+    return taskRunner;
+}
+
+// Recursively lists all files within the target folder, giving their full paths.
+function listFolderContents(folder):string[] {
+    var result:string[] = [];
+    var filesInFolder = fs.readdirSync(folder);
+
+    filesInFolder.forEach(function (fileInFolder) {
+        result.push(path.join(folder, fileInFolder));
+        if (fs.statSync(path.join(folder, fileInFolder)).isDirectory()) {
+            result = result.concat(listFolderContents(path.join(folder, fileInFolder)));
+        }
+    });
+
+    return result;
+}
+
+// Adds mock exist, checkPath, rmRF and mkdirP responses for given file paths.
+// Takes an object to add to and an array of file paths for which responses should be added.
+// Modifies and returns the argument object.
+function setupMockResponsesForPaths(responseObject:any, paths: string[]) { // Can't use rest arguments here (gulp-mocha complains)
+
+    // Create empty objects for responses only if they did not already exist (avoid overwriting existing responses)
+    responseObject.exist = responseObject.exist || {};
+    responseObject.checkPath = responseObject.checkPath || {};
+    responseObject.rmRF = responseObject.rmRF || {};
+    responseObject.mkdirP = responseObject.mkdirP || {};
+
+    var rmRFSuccessObj = {
+        success: true,
+        message: "foo bar"
+    };
+
+
+    paths.forEach((path) => {
+        responseObject.exist[path] = true;
+        responseObject.checkPath[path] = true;
+        responseObject.rmRF[path] = rmRFSuccessObj;
+        responseObject.mkdirP[path] = true;
+    });
+
+    return responseObject;
+}
+
+// Create temp dirs for mavencodeanalysis tests to save into
+function createTempDirsForCodeAnalysisTests():void {
```

Comment:
We ran into some errors with calls to the vsts-task-lib directly from tests, so I've replaced the much more convenient tl.rmRF() and tl.mkdirP() calls with a slightly clunkier system that only uses the default Node filesystem library.